Build failure against recent casacore

Build fails with

```
/usr/local/src/DP3/DPPP/DPRun.cc: In static member function 'static DP3::DPPP::DPStep::ShPtr DP3::DPPP::DPRun::makeSteps(const DP3::ParameterSet&, const string&, DP3::DPPP::DPInput*)':
/usr/local/src/DP3/DPPP/DPRun.cc:278:91: error: no matching function for call to 'casacore::DirectoryIterator::DirectoryIterator(casacore::Directory&, casacore::String)'
                                              casacore::Regex::fromPattern(path.baseName()));
                                                                                           ^
In file included from /usr/local/src/DP3/DPPP/DPRun.cc:64:
/usr/local/include/casacore/casa/OS/DirectoryIterator.h:117:5: note: candidate: 'casacore::DirectoryIterator::DirectoryIterator(const casacore::DirectoryIterator&)'
     DirectoryIterator (const DirectoryIterator& that);
     ^~~~~~~~~~~~~~~~~
/usr/local/include/casacore/casa/OS/DirectoryIterator.h:117:5: note:   candidate expects 1 argument, 2 provided
/usr/local/include/casacore/casa/OS/DirectoryIterator.h:113:5: note: candidate: 'casacore::DirectoryIterator::DirectoryIterator(const casacore::Directory&, const casacore::Regex&)'
     DirectoryIterator (const Directory& dir, const Regex& regExpression);
     ^~~~~~~~~~~~~~~~~
/usr/local/include/casacore/casa/OS/DirectoryIterator.h:113:5: note:   no known conversion for argument 2 from 'casacore::String' to 'const casacore::Regex&'
/usr/local/include/casacore/casa/OS/DirectoryIterator.h:108:5: note: candidate: 'casacore::DirectoryIterator::DirectoryIterator(const casacore::Directory&)'
     DirectoryIterator (const Directory& dir);
     ^~~~~~~~~~~~~~~~~
/usr/local/include/casacore/casa/OS/DirectoryIterator.h:108:5: note:   candidate expects 1 argument, 2 provided
/usr/local/include/casacore/casa/OS/DirectoryIterator.h:103:5: note: candidate: 'casacore::DirectoryIterator::DirectoryIterator()'
     DirectoryIterator();
     ^~~~~~~~~~~~~~~~~
/usr/local/include/casacore/casa/OS/DirectoryIterator.h:103:5: note:   candidate expects 0 arguments, 2 provided
```

This is against a pull from the casacore github.

--- a/DPPP/DPRun.cc
+++ b/DPPP/DPRun.cc
@@ -275,7 +275,7 @@ namespace DP3 {
             casacore::Directory dir(dirName);
             // Use the basename as the file name pattern.
             casacore::DirectoryIterator dirIter (dir,
-                                             casacore::Regex::fromPattern(path.baseName()));
+              casacore::Regex(casacore::Regex::fromPattern(path.baseName())));
             while (!dirIter.pastEnd()) {
               names.push_back (dirName + '/' + dirIter.name());
               dirIter++;

